src/snapshot/component_copy.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    GgrsComponentSnapshot, GgrsSnapshots, LoadWorld, LoadWorldSet, Rollback, RollbackFrameCount,
+    SaveWorld, SaveWorldSet,
+};
+use bevy::prelude::*;
+use std::marker::PhantomData;
+
+/// A [`Plugin`] which manages snapshots for a [`Component`] `C` using [`Copy`].
+pub struct GgrsComponentSnapshotCopyPlugin<C>
```

Comment:
What do we gain by having this one in addition to the clone-based one?

Reply:
Clone explicitly calls whatever cloning procedure the type implements, whereas Copy does a bitwise copy of the underlying memory. In theory, the Copy snapshot should be the absolute fastest without some domain specific tricks.

Reply:
Yeah according to the [Rust docs](https://doc.rust-lang.org/std/marker/trait.Copy.html#whats-the-difference-between-copy-and-clone), `Copy` will always do a bitwise `memcopy`, even if you have a `Clone` implementation that's completely different.

> Copies happen implicitly, for example as part of an assignment y = x. The behavior of Copy is not overloadable; it is always a simple bit-wise copy.

Which is why `Copy` is allowed to be implicit, since it's virtually free. I think the confusion might be coming from the requirement that anything which is `Copy` must also be `Clone`. Which you would think implies `Copy` uses `Clone`, but it actually doesn't.

Reply:
Having said that, for types which can be `Copy`, I think their `Clone` is (usually) derived to be `*self`, so potentially `Clone` is as fast as `Copy`, but it can be slower if the implementation isn't that.

Reply:
Don't think I've ever seen different Copy and Clone implementations.

But yeah, just wanted to make sure I didn't miss anything. So the value of this extra plugin is basically that the caller knows that the operation will be relatively cheap(-ish).